### PR TITLE
Integration of dynamic partial reconfiguration

### PR DESCRIPTION
**Pull request by biruk-belay**

### CODE REVIEW

File: rtl/sockets/csr/esp_tile_csr.vhd [hunk 1]
```diff
@@ -149,7 +150,7 @@ architecture rtl of esp_tile_csr is
     constant RESET_DCO_CFG : std_logic_vector(30 downto 0) := dco_reset_config_ovr;
 
     constant DEFAULT_CONFIG : std_logic_vector(ESP_CSR_WIDTH - 1 downto 0) :=
-        DEFAULT_ACC_COH & DEFAULT_DDR_CFG2 & DEFAULT_DDR_CFG1 & DEFAULT_DDR_CFG0 & DEFAULT_CPU_LOC_OVR & DEFAULT_ARIANE_HARTID &
+        "00" & DEFAULT_ACC_COH & DEFAULT_DDR_CFG2 & DEFAULT_DDR_CFG1 & DEFAULT_DDR_CFG0 & DEFAULT_CPU_LOC_OVR & DEFAULT_ARIANE_HARTID &
```
Comment (jzuckerman) [inline rtl/sockets/csr/esp_tile_csr.vhd:153]: can you define a constant here like for the other fields?

File: rtl/sockets/csr/esp_tile_csr.vhd [hunk 2]
```diff
@@ -183,7 +184,7 @@ architecture rtl of esp_tile_csr is
           case csr_addr is
             when ESP_CSR_VALID_ADDR =>
               readdata(ESP_CSR_VALID_MSB - ESP_CSR_VALID_LSB downto 0) <=
-                config_r(ESP_CSR_VALID_MSB downto ESP_CSR_VALID_LSB);
+               config_r(ESP_CSR_VALID_MSB downto ESP_CSR_VALID_LSB);
```
Comment (jzuckerman) [inline rtl/sockets/csr/esp_tile_csr.vhd:187]: undo this change

File: rtl/sockets/csr/esp_tile_csr.vhd [hunk 3]
```diff
@@ -270,6 +273,8 @@ architecture rtl of esp_tile_csr is
                 config_r(ESP_CSR_DDR_CFG2_MSB downto ESP_CSR_DDR_CFG2_LSB) <= apbi.pwdata(ESP_CSR_DDR_CFG2_MSB - ESP_CSR_DDR_CFG2_LSB downto 0);
               when ESP_CSR_ACC_COH_ADDR =>
                 config_r(ESP_CSR_ACC_COH_MSB downto ESP_CSR_ACC_COH_LSB) <= apbi.pwdata(ESP_CSR_ACC_COH_MSB - ESP_CSR_ACC_COH_LSB downto 0);
+              when ESP_CSR_ACC_DECOUPLER_ADDR =>
+                config_r(ESP_CSR_ACC_DECOUPLER_MSB downto ESP_CSR_ACC_DECOUPLER_LSB) <= apbi.pwdata(ESP_CSR_ACC_DECOUPLER_MSB - ESP_CSR_ACC_DECOUPLER_LSB downto 0); 
               when ESP_CSR_SRST_ADDR =>
                 srst <= wdata(0);
               when others => null;
```
Comment (jzuckerman) [inline rtl/sockets/csr/esp_tile_csr.vhd:280]: is the INTR read only and the DECOUPLER write only?

Comment (biruk-belay) [reply rtl/sockets/csr/esp_tile_csr.vhd:286]: INTR is read-only but DECOUPLER is rd/wr. But weirdly, reading it, even in this way, it returns the correct value. But I fixed it to rd/wr.

File: rtl/sockets/csr/esp_tile_csr.vhd (hunk 2 above)
Comment (biruk-belay) [reply rtl/sockets/csr/esp_tile_csr.vhd]: done

File: rtl/sockets/csr/esp_tile_csr.vhd (hunk 1 above)
Comment (biruk-belay) [reply rtl/sockets/csr/esp_tile_csr.vhd]: done

File: rtl/sockets/proxy/axislv2noc.vhd [hunk 4]
```diff
@@ -723,7 +800,7 @@ begin  -- rtl
 
   -- Update FSM state
   process (clk, rst)
-  begin  -- process
+  begin  -- pocess
     if rst = '0' then                   -- asynchronous reset (active low)
```
Comment (jzuckerman) [inline rtl/sockets/proxy/axislv2noc.vhd:804]: fix this lol

File: tools/socgen/socmap_gen.py [hunk 5]
```diff
@@ -94,6 +95,10 @@
 DDR_HADDR["ariane"] = 0x800
 DDR_HADDR["ibex"] = 0x800
 
+PBS_HADDR = dict()
+PBS_HADDR["leon3"] = 0x500
+PBS_HADDR["ariane"] = 0xA00
```
Comment (jzuckerman) [inline tools/socgen/socmap_gen.py:100]: does this mean the bitstreams for Ariane get loaded at 0xA0000000? how was that decided?

Comment (biruk-belay) [reply tools/socgen/socmap_gen.py:100]: In baremetal, yes. This address is decided randomly (on the lower addresses of the DDR just to be safe)

Comment (jzuckerman) [reply tools/socgen/socmap_gen.py:100]: How is the address done for Linux? or it that still TBD?

Comment (biruk-belay) [reply tools/socgen/socmap_gen.py:100]: In Linux, the partial bitstreams are moved into (a non-predefined) the kernel memory space by the driver. Then the mem address is associated with the index of the bistream by the runtime manger (a kernel module)

**Comment by jzuckerman** [comment]
Reopening...
